doc/source/data/faq.rst

```diff
@@ -0,0 +1,308 @@
+two core uses cases:
+
+* **ML (training) ingest:** Loading, preprocessing, and ingesting data into one or more
+  (possibly distributed) model trainers.
+* **batch inference:** Loading, preprocessing, and performing parallel batch
+  inference on data.
+
+We have designed the Datasets APIs, data model, execution model, and
+integrations with these use cases in mind, and have captured these use cases in
+large-scale nightly tests to ensure that we're hitting our scalability, performance,
+and efficiency marks for these use cases.
+
+See our :ref:`ML preprocessing docs <datasets-ml-preprocessing>` for more information on
+this positioning.
+
+What should I not use Ray Datasets for?
+=======================================
+
+Ray Datasets is not meant to be used for generic ETL pipelines (like Spark) or
+scalable data science (like Dask, Modin, or Mars). Datasets integrates with these
+frameworks, allowing for efficient exchange of distributed data partitions often with
+zero-copy.
+
+See our :ref:`ML preprocessing docs <datasets-ml-preprocessing>` for more information on
+how we see Ray Datasets fitting into a larger ML pipeline picture.
+
+For data loading for training, how does Ray Datasets compare to other solutions?
+================================================================================
+
+There are several ML framework-specific and general solutions for loading data into
+model trainers. Below, we summarize some advantages Datasets offers over these more
+specific ingest frameworks.
+
+Torch datasets (and data loaders)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* **Framework-agnostic:** Datasets is framework-agnostic and portable between different
+  distributed training frameworks, while
+  `Torch datasets <https://pytorch.org/docs/stable/data.html>`__ are specific to Torch.
+* **No built-in IO layer:** Torch datasets do not have an I/O layer for common file formats or in-memory exchange
+  with other frameworks; users need to bring in other libraries and roll this
+  integration themselves.
+* **Generic distributed data processing:** Datasets is more general: it can handle
+  generic distributed operations, including global per-epoch shuffling,
+  which would otherwise have to be implemented by stitching together two separate
+  systems. Torch datasets would require such stitching for anything more involved
+  than batch-based preprocessing, and does not natively support shuffling across worker
+  shards. See our
+  `blog post <https://www.anyscale.com/blog/deep-dive-data-ingest-in-a-third-generation-ml-architecture>`__
+  on why this shared infrastructure is important for 3rd generation ML architectures.
+* **Lower overhead:** Datasets is lower overhead: it supports zero-copy exchange between
+  processes, in contrast to the multi-processing-based pipelines of Torch datasets.
+
+TensorFlow datasets
+~~~~~~~~~~~~~~~~~~~
+
+* **Framework-agnostic:** Datasets is framework-agnostic and portable between different
```

Comment:
Many of these points are duplicated. Instead of grouping by competitor, list our strengths and mention how competitors are lacking in comparison.

Reply:
@clarkzinzow not sure how you want to handle this in the end, just saw @ericl 's comment on the subject, too.